swig/python/gdal-utils/osgeo_utils/gdal2tiles.py

```diff
@@ -1291,57 +1312,64 @@ def create_overview_tile(base_tz: int, base_tiles: List[Tuple[int, int]], output
             else:
                 tileposy = 0
 
+        if dsquerytile.RasterCount == tilebands - 1:
+            # assume that the alpha band is missing and add it
```

Comment:
in which situation can this happen ?

Reply:
I have seen it happen when there are no transparent areas in the base tile.. I will try to confirm it with a reproduction.

Reply:
and why is that a problem to not have an alpha band ? It looks to me as a desirable optimization to get smaller tiles

Reply:
The smaller tile size from the elision of the alpha band is desirable, it just plays foul with the overview tile creation code which assumes the presence of the alpha band in the base tiles, had to work around it by adding an opaque alpha band whenever we see only 3( or n - 1 ) bands in the base tile. I would have liked a cleaner way to work around this.

Reply:
ok, I missed that this was in the overview generation code path. Looks appropriate then